modules/ocf_kubernetes/manifests/master/loadbalancer.pp

```diff
@@ -114,16 +120,32 @@
 
   # Redirect from "cname.ocf.io" to "cname.ocf.berkeley.edu" for each cname.
   $cnames.each |$domain| {
+    $main_fqdns = prefix(['.ocf.io', ''], $domain)
+
+    # Get the list of aliases for this name, or an empty list if there are none
+    $extra_names = flatten([$extra_aliases[$domain]]).filter |$val| {
+      $val =~ NotUndef
+    }
+
+    # Append .ocf.berkeley.edu, .ocf.io, and empty string for each of the extra aliases
+    $extra_fqdns = flatten($extra_names.map |$name| {
+      prefix(['.ocf.berkeley.edu', '.ocf.io', ''], $name)
+    })
```

Comment:
Why not just hardcode these (and the `$extra_aliases` above) in an array or something instead? As this is, it's really hard to figure out what is going on, and hardcoding just a few domains would make it much more explicit and clear at the small cost of a bit of repetition.